src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,458 @@
+            return package_version_split[1]
+
+        if len(package_version_split) != 1:
+            self.composite_logger.log_error("[TDNF] Unexpected error during version epoch removal from package version. [PackageVersion={0}]".format(package_version))
+
+        return package_version
+
+    def get_package_size(self, output):
+        """Retrieve package size from installation output string"""
+        # Sample output line:
+        # Total download size: 15 M
+        if "Nothing to do" not in output:
+            lines = output.strip().split('\n')
+            for line in lines:
+                if line.find(self.STR_TOTAL_DOWNLOAD_SIZE) >= 0:
+                    return line.replace(self.STR_TOTAL_DOWNLOAD_SIZE, "")
+
+        return Constants.UNKNOWN_PACKAGE_SIZE
+    # endregion
+
+    # region auto OS updates
+    # endregion
+
+    # region Reboot Management
+    def is_reboot_pending(self):
+        """ Checks if there is a pending reboot on the machine. """
+        try:
+            pending_file_exists = os.path.isfile(self.REBOOT_PENDING_FILE_PATH)
+            pending_processes_exist = self.do_processes_require_restart()
+            self.composite_logger.log_debug("[TDNF] > Reboot required debug flags (tdnf): " + str(pending_file_exists) + ", " + str(pending_processes_exist) + ".")
+            return pending_file_exists or pending_processes_exist
+        except Exception as error:
+            self.composite_logger.log_error('[TDNF] Error while checking for reboot pending (tdnf): ' + repr(error))
+            return True  # defaults for safety
+
+    def do_processes_require_restart(self):
+        """Signals whether processes require a restart due to updates"""
+        self.composite_logger.log_verbose("[TDNF] Checking if process requires reboot")
+        # Checking using dnf-utils
+        code, out = self.env_layer.run_command_output(self.dnf_utils_prerequisite, False, False)  # idempotent, doesn't install if already present
+        self.composite_logger.log_verbose("[TDNF] Idempotent dnf-utils existence check. [Code={0}][Out={1}]".format(str(code), out))
+
+        # Checking for restart for distros with -r flag
+        code, out = self.env_layer.run_command_output(self.needs_restarting_with_flag, False, False)
+        self.composite_logger.log_verbose("[TDNF] > Code: " + str(code) + ", Output: \n|\t" + "\n|\t".join(out.splitlines()))
+        if out.find("Reboot is required") < 0:
+            self.composite_logger.log_debug("[TDNF] > Reboot not detected to be required (L1).")
+        else:
+            self.composite_logger.log_debug("[TDNF] > Reboot is detected to be required (L1).")
+            return True
+
+        # TODO: check if Double-checking using yum/zypper ps equivalent should be used?
+        # TODO: check what this command indicates: tdnf update --assumeno --reboot-required
```

Comment:
Marking todos on reboot.
Overall - please provide documentation being relied on for the behavior of the function overall.

Reply:
Pending

Reply:
Removed todos. Added ref doc in PR description/first comment